### ISSUE TITLE
Allow marking "model solutions"

### ISSUE THREAD

**Issue by niemela** [Kattis/problem-package-format#317]
This is a bit of a fringe feature. Most systems would ignore it, but it would not be much work to support.

A model solution is a solution that the authors would be happy to publish as a "good correct solution." It's probably a bit polished, possibly even commented.

I would suggest adding `model_solution: true` in `submissions.yaml`.

**Comment by eldering** [comment]
I have a feeling it would barely or not all be used to select submissions for publication (also because typically the package would get published as a whole), so even though I agree it's very cheap to add, I'd lean to not do so. No objections though if others indeed feel it's useful.

**Comment by Tagl** [comment]
I guess this is more of an educational feature that allows systems to identify these solutions with ease.

**Comment by evouga** [comment]
I agree with eldering; this feature would have been useful when contest directors needed to manually extract test data and judge solutions to post on contest web sites. It’s less useful if full packages are now on a centralized archive.

**Comment by niemela** [comment]
It is still useful in an educational setting. The difference between a "model solution" and any other "accepted" solution is that the former is intended to be polished, commented, and idiomatic , whereas the latter is just expected to work.

When publishing solutions after a contest we don't really care about polished, commented, and idiomatic, and contest code is seldom those things, but in an educational setting we do.

In any case, I'm not hearing any strong opposition to this?

**Comment by eldering** [comment]
I feel a bit that this is feature creep. If this is not used to publish a selected set of model submissions, will this then first of all be used by authors to mark these model solutions, and secondly by students to see which ones are? I doubt that.

**Comment by niemela** [comment]
> I feel a bit that this is feature creep. If this is not used to publish a selected set of model submissions, will this then first of all be used by authors to mark these model solutions, and secondly by students to see which ones are? I doubt that.

We already use something similar for education, so I would argue that I know that it would be used. Again, I don't think it will be used at contests much or at all.
